components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -1313,6 +1314,19 @@ private OutboundProvisioningConfig getOutboundProvisioningConfiguration(int appl
         return outBoundProvisioningConfig;
     }
 
+    private boolean isAuthenticationScriptConfigEnabledWithoutAdaptiveAuthenticationAvailable
+            (LocalAndOutboundAuthenticationConfig localAndOutboundAuthConfig) {
+        // return true if script based auth is enabled without adaptive auth is available
+        try {
+            AuthenticationScriptConfig authenticationScriptConfig = localAndOutboundAuthConfig
+                    .getAuthenticationScriptConfig();
+            return  !(ApplicationMgtUtil.isAdaptiveAuthenticationAvailable()) && authenticationScriptConfig.isEnabled();
+        } catch (NullPointerException nullPointerException) {
```

Comment:
Instead of catching NPE, find where it will be thrown and handle it at a fine grain level.